test/e2e/framework/job/wait.go

```diff
@@ -178,16 +195,28 @@ func WaitForJobFinishWithTimeout(ctx context.Context, c clientset.Interface, ns,
 	})
 }
 
-func isJobFinished(j *batchv1.Job) bool {
+func isJobFailed(j *batchv1.Job) bool {
+	return isConditionTrue(j, batchv1.JobFailed)
+}
+
+func isJobCompleted(j *batchv1.Job) bool {
+	return isConditionTrue(j, batchv1.JobComplete)
+}
+
+func isConditionTrue(j *batchv1.Job, condition batchv1.JobConditionType) bool {
 	for _, c := range j.Status.Conditions {
-		if (c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed) && c.Status == v1.ConditionTrue {
+		if c.Type == condition && c.Status == v1.ConditionTrue {
 			return true
 		}
 	}
 
 	return false
 }
 
+func isJobFinished(j *batchv1.Job) bool {
```

Comment:
nit: move the function above  isJobFailed so that the functions chains follow the pattern in file